### PR TITLE
Lessons from go 1.13

### PR DESCRIPTION
**Pull request by taylor-sutton**
* While pulling, I hit Github's Abuse Rate Limit => Now pull checks for that and waits if necessary.
* Clarify the messaging around pull as well - I found `WARNING: Github API timed out before completing query.` quite confusing as to what was going on.
* When there's a PR-not-approved error message, it's helpful to notify the user that there's a way to override this error.

### CODE REVIEW

File: initialize/initialize.go [hunk 1]
```diff
@@ -133,7 +134,17 @@ func githubSearch(query string) ([]Repo, error) {
 	numProcessedResults := 0
 	for {
 		result, resp, err := client.Search.Code(context.Background(), query, opts)
-		if err != nil {
+		if abuseErr, ok := err.(*github.AbuseRateLimitError); ok {
+			var waitTime time.Duration
+			if abuseErr.RetryAfter != nil {
+				waitTime = *abuseErr.RetryAfter
```
Comment (nathanleiby) [inline initialize/initialize.go]: 👍

File: initialize/initialize.go [hunk 2]
```diff
@@ -145,7 +156,7 @@ func githubSearch(query string) ([]Repo, error) {
 
 		incompleteResults := result.GetIncompleteResults()
 		if incompleteResults {
-			log.Println("WARNING: Github API timed out before completing query")
+			log.Println("WARNING: Github API timed out before completing query. Querying again for additional results.")
```
Comment (nathanleiby) [inline initialize/initialize.go]: I'm confused by this behavior and log message. I would think `incompleteResults` just means that we need to keep paginating... can we just remove this WARNING statement altogether? The next log line explains that we're paginating

https://godoc.org/github.com/google/go-github/github#IssuesSearchResult

File: merge/merge.go [hunk 3]
```diff
@@ -88,11 +88,11 @@ func GitHubMerge(ctx context.Context, input Input, repoLimiter *time.Ticker, mer
...
-			return Output{Success: false}, fmt.Errorf("PR awaiting review")
+			return Output{Success: false}, fmt.Errorf("PR awaiting review. Use --ignore-review-approval to override this check.")
 		}
 		for _, r := range reviews {
 			if r.GetState() != "APPROVED" {
-				return Output{Success: false}, fmt.Errorf("PR is not approved. Review state is %s", r.GetState())
+				return Output{Success: false}, fmt.Errorf("PR is not approved. Review state is %s. Use --ignore-review-approval to override this check.", r.GetState())
 			}
 		}
 	}
```
Comment (nathanleiby) [inline merge/merge.go]: relatedly, could also improve line 82 to say that

`--ignore-build-status      Ignore whether or not builds are passing`

lets you override

https://github.com/Clever/microplane/pull/50/files#diff-24cc3f380ea342a2744dcb5c93f4b2adR82

File: initialize/initialize.go (hunk 2 above)
Comment (taylor-sutton) [reply initialize/initialize.go]: I was trying to make it less confusing with the extra words, but your solution of just removing that line is better.

File: merge/merge.go (hunk 3 above)
Comment (taylor-sutton) [reply merge/merge.go]: 👍

**Pull request merged**